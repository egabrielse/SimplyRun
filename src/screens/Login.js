import React, { Component } from 'react';
import { Text, View, TextInput, Button, Alert  } from 'react-native';
import firebase from 'firebase';
import firebaseConfig from '../config/firebaseConfig'
import { connect } from 'react-redux'

//References to the root of the firestore database
const firestore = firebase.firestore();
//Firebase initialzation 
firebaseConfig

//Login Screen 
class Login extends Component {
    //Using the Firebase built in authentification to login 
    login = () => {
        if (this.props.email != null && this.props.password != null &&
            this.props.email.trim() != "" && this.props.password.trim() != "") {
            firebase
                .auth()
                .signInWithEmailAndPassword(this.props.email, this.props.password)
                .then(() => this.props.navigation.navigate('Main'))
                .catch(error => Alert.alert(error.message));
        }
    }
    //Function to change to the Create Acount Screen
    goToCreateAccount = () => {
        this.props.navigation.navigate('CreateAccount');
    }

    //Basic Funtion that sends some hard coded data to a specific place in the 
    //firestore database
    sendToFirebase = () => {
        firestore.collection('users').doc('test').set({
            name: "SimpleTest",
            weight: "260",
            height: "6"
        })
    }
    //Basic function that retrieves some date from firestore and displays it on the console
    getFromFirebase = () => {
        var ref = firestore.collection('users').doc('test');
        ref.get().then(testData => {
            console.log(testData.data())
        })
    }

    devSkip = () => {
        this.props.navigation.navigate('Main')
    }

    render() {
        return (
            <View style={{ flex: 1, alignContent: 'center', justifyContent: 'center' }} >
                <Text>Login</Text>
                <TextInput
                    placeholder="Email"
                    autoCapitalize="none"
                    onChangeText={email => this.props.typeEmail(email)}
                    value={this.props.email}
                />
                <TextInput
                    secureTextEntry
                    placeholder="Password"
                    autoCapitalize="none"
                    onChangeText={password => this.props.typePassword(password)}
                    value={this.props.password}
                />
                <Button title="Login" onPress={this.login} />
                <Button title="CreateAccount" onPress={this.goToCreateAccount} />
                <Text>This sends some hardcoded data to the firestore database</Text>
                <Button title="Write To Firebase" onPress={this.sendToFirebase} />
                <Text>This retrieves some data from firestore and prints it on the console </Text>
                <Button title="Read From Firebase" onPress={this.getFromFirebase} />
                <Button title="DEV:skip login" onPress={this.devSkip}/>

            </View>
        );
    }
    
}
//Getting the states from the store and mapping it to props in the Login
function mapStateToProps(state) {
    return {
        email: state.email,
        password: state.password
    }
}
//Sends actions to the reducer in the App.js file 
function mapDispatchtoProps(dispatch) {
    return {
        typeEmail: (email) => dispatch({ type: "EMAIL", email }),
        typePassword: (password) => dispatch({ type: "PASSWORD", password })
    }
}

//Connecting the react components to the store in App.js 
export default connect(mapStateToProps, mapDispatchtoProps)(Login);