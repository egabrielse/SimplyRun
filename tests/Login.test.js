// __tests__/Intro-test.js
import React from 'react';
import Login from '../src/screens/Login';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store'
import initialState from './mock_stores/initialState'
import { Provider } from 'react-redux';

const createMockStore = configureStore([])


describe("Login screen", () => {

    beforeEach(() => {
        mockStore = createMockStore(initialState)
    })

    it('renders correctly', () => {
        const tree = renderer.create(
            <Provider store={mockStore}>
                <Login/>
            </Provider>
        ).toJSON();
        expect(tree).toMatchSnapshot();
      });
})
