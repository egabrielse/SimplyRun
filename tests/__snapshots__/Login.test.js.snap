// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login screen renders correctly 1`] = `
<RCTScrollView
  collapsable={false}
  contentContainerStyle={
    Object {
      "flexGrow": 1,
    }
  }
  keyboardShouldPersistTaps="handled"
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
>
  <View>
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "flex": 1,
            "marginHorizontal": 20,
          },
          Object {
            "paddingBottom": 0,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 40,
              "fontStyle": "italic",
              "fontWeight": "bold",
            }
          }
        >
          Simply Run
        </Text>
      </View>
      <View
        style={
          Object {
            "flex": 2,
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          keyboardType="email-address"
          onChangeText={[Function]}
          onSubmitEditing={[Function]}
          placeholder="Email"
          rejectResponderTermination={true}
          returnKeyType="next"
          style={
            Object {
              "borderColor": "lightgrey",
              "borderWidth": 1,
              "height": 50,
              "justifyContent": "center",
              "maxHeight": 50,
              "padding": 8,
            }
          }
          underlineColorAndroid="transparent"
          value={null}
        />
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          keyboardType="email-address"
          onChangeText={[Function]}
          onSubmitEditing={[Function]}
          placeholder="Password"
          rejectResponderTermination={true}
          returnKeyType="done"
          secureTextEntry={true}
          style={
            Object {
              "borderColor": "lightgrey",
              "borderWidth": 1,
              "height": 50,
              "justifyContent": "center",
              "maxHeight": 50,
              "padding": 8,
            }
          }
          underlineColorAndroid="transparent"
          value={null}
        />
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "lightgray",
                "height": 50,
                "justifyContent": "center",
                "paddingHorizontal": 15,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "black",
                  "fontSize": 20,
                }
              }
            >
              Sign In!
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 75,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 18,
              }
            }
          >
            Don't have an account?
          </Text>
          <View
            accessibilityRole="button"
            accessibilityStates={Array []}
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {},
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#007AFF",
                      "fontSize": 18,
                      "padding": 8,
                      "textAlign": "center",
                    },
                  ]
                }
              >
                Sign Up!
              </Text>
            </View>
          </View>
        </View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "red",
                "height": 50,
                "justifyContent": "center",
                "maxHeight": 50,
                "padding": 8,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 20,
                }
              }
            >
              DEV:Skip Login
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
